File: sdk/java/src/main/java/com/aligntech/sdk/ConfigStore.java
package com.aligntech.sdk;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;


class ConfigStore {

    private volatile Map<String, FlagConfig> flags = new ConcurrentHashMap<>();
    private volatile String version;

    void replaceAll(String version, Map<String, FlagConfig> newFlags) {
        this.version = version;
        this.flags = new ConcurrentHashMap<>(newFlags);
    }

    FlagConfig get(String flagKey) {
        return flags.get(flagKey);
    }

    String version() {
        return version;
    }

    int size() {
        return flags.size();
    }

    boolean isEmpty() {
        return flags.isEmpty();
    }
}

File: sdk/javascript/src/ConfigStore.ts
import { FlagConfig } from './types';

export class ConfigStore {
  private flags: Map<string, FlagConfig> = new Map();
  private _version: string = '';

  replaceAll(version: string, newFlags: Map<string, FlagConfig>): void {
    this._version = version;
    this.flags = new Map(newFlags);
  }

  get(flagKey: string): FlagConfig | undefined {
    return this.flags.get(flagKey);
  }

  get version(): string {
    return this._version;
  }

  get size(): number {
    return this.flags.size;
  }

  isEmpty(): boolean {
    return this.flags.size === 0;
  }
}
